### PR TITLE
Refactor Library

### PR DESCRIPTION
**Pull request by RodriSanchez1**

### CODE REVIEW

File: src/engine.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+};
+
+async function fetchPictogramsURLs(
+  words: string[],
+  symbolSet: string,
+  language: string
+): Promise<Pictogram[]> {
+  try {
+    const requests = words.map((word) =>
+      axios.get(globalSymbolsURL, {
```
Comment (maxicapodacqua) [inline src/engine.ts:95]: Maybe in a future iteration use `fetch` instead of `axios` as now it's included in v8 and node

File: src/engine.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
+import { Configuration, OpenAIApi } from "openai";
+import axios, { AxiosRequestConfig } from "axios";
+import { txt2imgBody } from "./request";
+import { ConfigurationParameters } from "openai";
+
+let openaiInstance: OpenAIApi;
+let globalSymbolsURL = "https://www.globalsymbols.com/api/v1/labels/search/";
+let pictonizerURL = "";
```
Comment (maxicapodacqua) [inline src/engine.ts:8]: Consider using a global config object instead of multiple global variables

File: src/engine.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
+import { Configuration, OpenAIApi } from "openai";
+import axios, { AxiosRequestConfig } from "axios";
+import { txt2imgBody } from "./request";
+import { ConfigurationParameters } from "openai";
+
+let openaiInstance: OpenAIApi;
+let globalSymbolsURL = "https://www.globalsymbols.com/api/v1/labels/search/";
```
Comment (maxicapodacqua) [inline src/engine.ts:7]: Use a `const` for this default value

File: src/engine.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    return pictogramsList;
+  } catch (error: Error | any) {
+    throw new Error("Error fetching pictograms URLs " + error.message);
+  }
+}
+
+async function pictonizer(imagePrompt: string): Promise<string> {
+  //TODO pedir que me de un prompt mejorado del picto a mejorar
+  //https://platform.openai.com/docs/guides/prompt-engineering
+  //https://learnprompting.org/es/docs/intro Imageprompting
```
Comment (maxicapodacqua) [inline src/engine.ts:134]: Add a tech debt issue for this

File: src/engine.ts [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+      };
+      return JSON.stringify(resultJson);
+    }
+  } catch (error: Error | any) {
+    console.error("Error generating pictogram: ", error.message);
+  } finally {
+    return JSON.stringify({
+      images: [{ data: "ERROR Generating Pictogram" }],
+      prompt: imagePrompt,
+    });
```
Comment (maxicapodacqua) [inline src/engine.ts:183]: Move this to the catch block, `finally` should be used to release resources

File: src/engine.ts [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+async function processPictograms(pictogramsURL: Pictogram[]) {
+  const updatedPictograms = await Promise.all(
+    pictogramsURL.map(async (pictogram) => {
+      const id = parseInt(pictogram.id);
+      if (isNaN(id)) {
+        return {
+          ...pictogram,
+          id: 123456, //TODO add library to get id nanoid
```
Comment (maxicapodacqua) [inline src/engine.ts:194]: Add tech debt issue for this

File: src/request.ts
Comment (maxicapodacqua) [inline src/request.ts:1]: Maybe rename this file? I'm not sure what are these constants used for. If the file is only for declaring constants use `constants.ts`

**Review by maxicapodacqua — CHANGES_REQUESTED** [review]

File: src/engine.ts [hunk 7]
```diff
@@ -0,0 +1,236 @@
+import { Configuration, OpenAIApi } from "openai";
+import axios, { AxiosRequestConfig } from "axios";
+import { txt2imgBody } from "./request";
+import { ConfigurationParameters } from "openai";
+
+let openaiInstance: OpenAIApi;
```
Comment (shayc) [inline src/engine.ts:6]: `openAIInstance`, for consistency across other variables in this file where open AI is used.

File: src/engine.ts [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+
+    const pictogramsList: Pictogram[] = dataList.map((data) => ({
+      id: data[0].id,
+      text: data[0].text,
+      locale: data[0].language,
+      picto: data.map((picto: any) => picto.picto.image_url),
+    }));
+
+    return pictogramsList;
+  } catch (error: Error | any) {
```
Comment (shayc) [inline src/engine.ts:126]: Is `any` needed here?

File: src/request.ts
Comment (RodriSanchez1) [reply src/request.ts:1]: Done

File: src/engine.ts (hunk 8 above)
Comment (RodriSanchez1) [reply src/engine.ts:119]: I get `Catch clause variable type annotation must be 'any' or 'unknown' if specified.` if delet it. Which one is fits better in this case? `any` or `unknow`

File: src/engine.ts [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+              picto: { image_url: "ERROR: No image in the Symbol Set" },
+            },
+          ]
+    );
+
+    const pictogramsList: Pictogram[] = dataList.map((data) => ({
+      id: data[0].id,
+      text: data[0].text,
+      locale: data[0].language,
+      picto: data.map((picto: any) => picto.picto.image_url),
```
[the remark this answers is not in the document]
Comment (RodriSanchez1) [reply src/engine.ts:115]: How can i deal with this, if the data is a third party API response? I will create an issue of this

File: src/engine.ts (hunk 4 above)
Comment (RodriSanchez1) [reply src/engine.ts:127]: @hectoritr Can you create this issue?

File: src/engine.ts (hunk 6 above)
Comment (RodriSanchez1) [reply src/engine.ts:185]: Done #4

File: src/engine.ts [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+
+    return wordsSuggestionsList;
+  }
+  return [];
+}
+
+type Pictogram = {
+  id: string;
+  text: string;
+  locale: string;
```
[the remark this answers is not in the document]
Comment (RodriSanchez1) [reply src/engine.ts:77]: I will do it in other PR #5

**Pull request merged**